mr/src/main/java/org/elasticsearch/hadoop/rest/RestService.java

```diff
@@ -256,6 +256,9 @@ public void remove() {
                 }
             }
             final Map<String, NodeInfo> nodesMap = new HashMap<String, NodeInfo>();
+            if (nodes == null){
+                nodes = client.getRestClient().getHttpNodes(false);
```

Comment:
This method *is* the discovery process. We shouldn't be discovering nodes as a work around to null pointers on disabled discovery. I think a better option might be to simply return an empty list on line 218 instead of null.